theia-ide/theia
# [plugin] Implement SCM Repository selected event

#### What it does

Implement SCM Repository selected event: https://github.com/microsoft/vscode/blob/a50cf110d96fbe50fc460a6de14ade5b6b060dd4/src/vs/vscode.proposed.d.ts#L673-L684
Removed duplicated `SourceControlResourceDecorations` interface.
#### How to test

1. Open a workspace with more than one Git repositories.
2. Clone to the workspace folder and compile the test plugin: https://github.com/vinokurig/Test.git
3. Start hosted instance with the plugin.
4. Select the `/` SCM repository (this is a view of a dummy SCM repository):
![screenshot-localhost-3030-2019 09 06-17-34-24](https://user-images.githubusercontent.com/7668752/64436695-892fe000-d0cd-11e9-8f13-ccd7492bcbb7.png)

See the notification that is fired inside the event handler.
Select other repository again.
See the notification that the repository is **NOT** selected.
#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
